Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs

```diff
@@ -26,204 +30,174 @@ public class CodeGenerationDisplayDriver : ContentTypeDefinitionDisplayDriver
+                case JObject jObject:
+                    // Using a quoted string so it doesn't mess up the syntax highlighting of the rest of the code.
+                    return T["\"FIX ME! Couldn't determine the actual type to instantiate.\" {0}", jObject.ToString()];
+                default:
+                    throw new NotSupportedException($"Settings values of type {jToken.GetType()} are not supported.");
+            }
+        }
+
+        private void AddFilteredSettings<T>(StringBuilder codeBuilder, JObject settings, int indentationDepth)
+        {
+            var indentation = string.Join(string.Empty, Enumerable.Repeat(" ", indentationDepth));
+
+            var filteredSettings = ((IEnumerable<KeyValuePair<string, JToken>>)settings)
+                .Where(setting => setting.Key != typeof(T).Name);
+            foreach (var setting in filteredSettings)
+            {
+                var properties = setting.Value.Where(property => property is JProperty).Cast<JProperty>().ToArray();
+
+                if (properties.Length == 0) continue;
+
+                codeBuilder.AppendLine($"{indentation}.WithSettings(new {setting.Key}");
+                codeBuilder.AppendLine(indentation + "{");
+
+                // This doesn't support multi-level object hierarchies for settings but come on, who uses
+                // complex settings objects?
+                for (int i = 0; i < properties.Length; i++)
+                {
+                    var property = properties[i];
+                    codeBuilder.AppendLine($"{indentation}    {property.Name} = {ConvertJToken(property.Value)}" +
+                                           $"{(i != properties.Length - 1 ? "," : string.Empty)}");
+                }
+
+                codeBuilder.AppendLine(indentation + "})");
+            }
+        }
+
+
+        private static void GenerateCodeForSettings(StringBuilder codeBuilder, ContentTypeSettings contentTypeSettings)
+        {
+            if (contentTypeSettings.Creatable) codeBuilder.AppendLine("    .Creatable()");
+            if (contentTypeSettings.Listable) codeBuilder.AppendLine("    .Listable()");
+            if (contentTypeSettings.Draftable) codeBuilder.AppendLine("    .Draftable()");
+            if (contentTypeSettings.Versionable) codeBuilder.AppendLine("    .Versionable()");
+            if (contentTypeSettings.Securable) codeBuilder.AppendLine("    .Securable()");
+            if (!string.IsNullOrEmpty(contentTypeSettings.Stereotype))
+            {
+                codeBuilder.AppendLine($"    .Stereotype(\"{contentTypeSettings.Stereotype}\")");
+            }
+        }
+
+        private static void AddWithLine(StringBuilder codeBuilder, string name, string value)
+        {
+            if (!string.IsNullOrEmpty(value))
+            {
+                codeBuilder.AppendLine($"        .With{name}(\"{value}\")");
+            }

```

Comment:
Lehetne egy sor, és akkor expression-bodied.

Reply:
Hogy? Az if akkor is statement ha csak egy sor.

Reply:
Ja tényleg.